### PR TITLE
Remote config control for circuits feature

### PR DESCRIPTION
**Pull request by kriskowal**
Alternate solution to circuit enable/disable that uses just one flag.

r @jcorbin @raynos @ShanniLi

### CODE REVIEW

**Comment by Raynos** [comment]
:-1: I've seen bugs with the `LockedHealthy` state in production.

File: node/hyperbahn/service_proxy.js [hunk 1]
```diff
@@ -252,7 +258,7 @@ function createServiceChannel(serviceName) {
     var handler = new RelayHandler(svcchan);
 
     // Decorate a circuit health monitor to egress request handlers.
-    if (mode === 'exit' && self.circuits) {
+    if (mode === 'exit') {
```
Comment (Raynos) [inline node/hyperbahn/service_proxy.js]: Let's keep the if here.

File: node/hyperbahn/service_proxy.js [hunk 2]
```diff
@@ -536,6 +540,18 @@ function destroy() {
     self.rateLimiter.destroy();
 };
 
+ServiceDispatchHandler.prototype.enableCircuits =
+function enableCircuits() {
```
Comment (Raynos) [inline node/hyperbahn/service_proxy.js]: Let's mutate all the exit handlers here.

File: node/hyperbahn/service_proxy.js [hunk 3]
```diff
@@ -536,6 +540,18 @@ function destroy() {
...
+    var self = this;
+    self.circuits.reset(states.HealthyState);
+};
+
+ServiceDispatchHandler.prototype.disableCircuits =
+function disableCircuits() {
```
Comment (Raynos) [inline node/hyperbahn/service_proxy.js]: Let's mutate all the exit handlers here.

Comment (ShanniLi) [inline node/hyperbahn/service_proxy.js]: +1, the current approach is a little confusing.

**Comment by kriskowal** [comment]
Alternate solution, now atop the simplified circuit thru relay code @jcorbin has since landed, that turns enables and disables circuits with a single flag on the service dispatch handler, to which relays refer.

r @jcorbin @rf

**Comment by jcorbin** [comment]
lgtm

**Pull request merged**